app/components/Carousel.tsx

```diff
@@ -0,0 +1,128 @@
+import React from "react"
+import { View, FlatList, Dimensions, ViewStyle, ImageSourcePropType, TextStyle } from "react-native"
+import { CarouselCard, Text } from "../components"
+import { spacing } from "../theme"
+import Animated, { useAnimatedScrollHandler, useSharedValue } from "react-native-reanimated"
+import { openLinkInBrowser } from "../utils/openLinkInBrowser"
+import { openMap } from "../utils/openMap"
+
+interface StaticCarouselProps {
+  preset: "static"
+  data: ImageSourcePropType[]
+  subtitle: string
+  meta?: string
+  body: string
+}
+
+interface ButtonData {
+  text: string
+  link: string
+}
+
+export interface DynamicCarouselItem {
+  image: ImageSourcePropType
+  subtitle: string
+  meta?: string
+  body: string
+  leftButton?: ButtonData
+}
+
+interface DynamicCarouselProps {
+  preset: "dynamic"
+  data: DynamicCarouselItem[]
+}
+
+type CarouselProps =
+  | (StaticCarouselProps | DynamicCarouselProps) & {
+      title: string
+      openLink?: () => void
+    }
+
+export const CAROUSEL_IMAGE_WIDTH = Dimensions.get("screen").width - spacing.medium
+
+// ! https://github.com/software-mansion/react-native-reanimated/issues/457
+const AnimatedFlatList = Animated.createAnimatedComponent(FlatList)
+
+const openLink = (destination: string) =>
+  destination.startsWith("https") ? openLinkInBrowser(destination) : openMap(destination)
```

Comment:
would recommend that we put this into both `openLinkInBrowser` and `openMap` so we don't have to check for security every time 😃

Reply:
I agree with JM

Also maybe this should live in the `utils` folder? I can see other tabs needing to open a link

Reply:
this will likely change when we understand the shape of the data coming back from webflow so will just leave it for now.